Return field names in the scoring response by default.

osery opened the pull request:
This can be disabled by passing in `"noFieldNames": true` in the scoring request and works both for AWS lambda and local rest scorer.

This is a relatively safe option as it adds a new field that can be easily ignored by existing clients. There is still a small chance of breaking an existing client who might have strict JSON parsing enabled. E.g., for Java using `FAIL_ON_UNKNOWN_PROPERTIES` (https://fasterxml.github.io/jackson-databind/javadoc/2.6/com/fasterxml/jackson/databind/DeserializationFeature.html#FAIL_ON_UNKNOWN_PROPERTIES)

**Q:** Why not build flag or java property?
**A:** These would apply globally for all requests. I think it is better when we give users the option to switch between more and less verbose on the fly as needed. This allows low-bandwidth when working with production queries as well as human debugging, when needed, without restarting the scorer.

mmalohlava commented on aws-lambda-scorer/README.md:
````diff
@@ -110,6 +110,67 @@ The expected response should follow this structure, but the actual values may di
+```
+
+To get back also the names of fields in the order in which they appear in the resulting `score`,
+pass in an extra request field `includeFieldNames` set to `true`.
````
Note: should it be called `includeClassNamesInOutput`?

osery replied:
The way it is implemented, it turns names for all the elements in the output. This includes both classes and any fields copied from the input, plus optionally the is field. Wdyt?

nkpng2k replied:
I'm tempted to say that this should be `true` by default, and `false` only by request... would invite less confusion to first time users, and realistically should not be costly enough where it ever needs to be ignored except in special cases...

osery replied:
Thanks @nkpng2k, done.

nikhilgk replied:
> Note: should it be called `includeClassNamesInOutput`?
> 
Since this change was done, shouldn't the JSON response field name be "class" or "targetClass" instead of "fields"?

osery replied:
I implemented the content regarding the default value and tried to explain why the "class" weird does not match. Please see my other comment below :+1:.

tomkraljevic commented:
i would also say enable by default.

osery commented:
Changed to make returning the fields the default. Inverted and renamed the flag as "noFieldNames" so that `false` and missing the flag have the same behavior (both are JSON falsy).

I kept the name derived for "field" (as opposed to "class"). This is consistent with the input request field `fields` (which I don't dare changing). Moreover, it also contains possibly more values than resulting class names (e.g., names of fields copied from the input and possibly also the `idField` name).

PTAL.

nikhilgk commented on aws-lambda-scorer/README.md:
````diff
@@ -110,6 +110,13 @@ The expected response should follow this structure, but the actual values may di
 ```json
 {
+  "id": "a12e7390-b8ac-406a-ade9-0d5ea4b63ea9",
+  "fields": [
````
Should we change this to "class" or "classNames" or "targetClasses"?

osery replied:
Please read my other comments. This is not only classes and there is also consistency with the input field.

nkpng2k replied:
agree with @osery here... `fields` encompasses all header values returned, not just the mojo generated classes from target.

tomkraljevic commented:
the specific word “class” is a perhaps a poor choice since it is a reserved keyword in many languages, including java.

osery commented:
Discussed with @mmalohlava offline and decided to also rename `noFieldNames` to `noFieldNamesInOutput` for consistency with the `includeFieldsInOutput`.

osery commented:
I don't see any blocking comments, so merging now.
